### PR TITLE
fix: use env vars when authenticating with docker

### PR DESCRIPTION
**Pull request by garethjevans**

### CODE REVIEW

**Review by slide — CHANGES_REQUESTED** [review]
We had something like this before and it caused issues because of the credentials on the ci server.

**Comment by slide** [comment]
I think this should have a flag involved if you want wanting to use this script from your own system. I don't think we should merge this as is.

**Comment by garethjevans** [comment]
This was based on the current master failures of:

```
./.ci/publish-tags.sh: line 30: DOCKERHUB_USERNAME: unbound variable
```

I took a look at the other scripts in the `.ci` folder and they all seem to use this method of authenticating.

**Comment by olblak** [comment]
wouldn't it be easier to set DOCKERHUB_USERNAME and DOCKERHUB_PASSWORD on ci.j.io and trusted.ci? at the moment we use a config.json

**Comment by slide** [comment]
The publish-tags script needs to be fixed to not do this.

**Comment by garethjevans** [comment]
@olblak @slide happy to go with whatever you think is best.  I can change this PR to use the variables if you think thats best

**Comment by slide** [comment]
We already have DOCKER_CONFIG being set on ci.j.io and trusted. So, I think this `docker login` call should be wrapped with a check if DOCKER_CONFIG is empty or not set.

https://github.com/jenkins-infra/pipeline-library/blob/5b6efab292955b7cd73760a4cc9d4546f5f50525/vars/infra.groovy#L18

File: .ci/publish-images.sh [hunk 1]
```diff
@@ -27,7 +27,11 @@ if [[ "$DOCKERHUB_ORGANISATION" == "jenkins" ]]; then
 fi
 
 docker-login() {
-    docker login --username ${DOCKERHUB_USERNAME} --password ${DOCKERHUB_PASSWORD}
+    if [ -z "$DOCKER_CONFIG" ]; then
+      docker login --username ${DOCKERHUB_USERNAME} --password ${DOCKERHUB_PASSWORD}
+    else
+      docker login
```
Comment (slide) [inline .ci/publish-images.sh:33]: I don't think we want this at all, I believe docker will automatically look for DOCKER_CONFIG and use it.

**Comment by garethjevans** [comment]
@olblak @slide I'm going to close this PR until we can get some consensus on a way forward.

**Pull request closed without merging**